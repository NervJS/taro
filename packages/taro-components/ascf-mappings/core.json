{
  "Button": {
    "size": {
      "ascfProp": "dimension",
      "valueMap": {
        "mini": "XS",
        "small": "S",
        "default": "M",
        "large": "L"
      },
      "required": false,
      "default": "M",
      "docLink": "https://ascf.docs/components/button#dimension"
    },
    "loading": {
      "ascfProp": "isLoading",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/button#isloading"
    }
  },
  "Input": {
    "password": {
      "ascfProp": "secretInput",
      "type": "boolean",
      "required": false,
      "default": false,
      "docLink": "https://ascf.docs/components/input#secretinput"
    }
  }
}
