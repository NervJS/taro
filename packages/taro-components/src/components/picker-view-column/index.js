import Nerv from 'nervjs'

class PickerViewColumn extends Nerv.Component {
  componentDidMount () {
    console.error('H5 暂不支持 PickerViewColumn 组件！')
  }

  render () {
    const { ...rest } = this.props
    return <div {...rest}>{this.props.children}</div>
  }
}

export default PickerViewColumn
