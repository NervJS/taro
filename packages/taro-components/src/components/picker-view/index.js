import Nerv from 'nervjs'

class PickerView extends Nerv.Component {
  componentDidMount () {
    console.error('H5 暂不支持 PickerView 组件！')
  }

  render () {
    const { ...reset } = this.props
    return <div {...reset}>{this.props.children}</div>
  }
}

export default PickerView
