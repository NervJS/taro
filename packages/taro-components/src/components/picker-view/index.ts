export * from './picker-view-column'
export * from './picker-view'
