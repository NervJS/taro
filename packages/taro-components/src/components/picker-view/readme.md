# taro-picker-view-core



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute         | Description | Type       | Default     |
| ---------------- | ----------------- | ----------- | ---------- | ----------- |
| `indicatorClass` | `indicator-class` |             | `string`   | `undefined` |
| `indicatorStyle` | `indicator-style` |             | `string`   | `undefined` |
| `maskClass`      | `mask-class`      |             | `string`   | `undefined` |
| `maskStyle`      | `mask-style`      |             | `string`   | `undefined` |
| `value`          | --                |             | `number[]` | `undefined` |


## Events

| Event       | Description | Type               |
| ----------- | ----------- | ------------------ |
| `change`    |             | `CustomEvent<any>` |
| `pickend`   |             | `CustomEvent<any>` |
| `pickstart` |             | `CustomEvent<any>` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
