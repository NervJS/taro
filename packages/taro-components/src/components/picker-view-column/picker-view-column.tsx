// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { Component, h, Host } from '@stencil/core'

@Component({
  tag: 'taro-picker-view-column-core'
})
export class PickerViewColumn {
  componentDidLoad () {
    console.error('H5 暂不支持 PickerViewColumn 组件！')
  }

  render () {
    return (
      <Host />
    )
  }
}
