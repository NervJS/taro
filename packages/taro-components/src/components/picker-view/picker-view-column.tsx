import { Component, h, Host, Listen, Element, Event, EventEmitter, Prop, State } from '@stencil/core'

import { debounce } from '../../utils'

@Component({
  tag: 'taro-picker-view-column-core',
  styleUrl: './style/column.scss'
})
export class PickerViewColumn {
  // 当前 ref
  @Element() el: HTMLElement

  // 标记属于 picker 第几列
  @Prop() col: string

  // 初始化的选中位置
  @Prop({ attribute: 'initial-position' }) initialPosition = '0'

  // 滑动距离上下留白区域-通过父视图和 indicator 计算而来
  @Prop({ attribute: 'padding-vertical' }) paddingVertical = 0

  @State()
  isInit: boolean = false

  @State()
  isMove: boolean = false

  // 选中后的结果回调
  @Event({ eventName: 'onselect' }) onChange: EventEmitter
  @Event({ eventName: 'onselectstart' }) onSelectStart: EventEmitter
  @Event({ eventName: 'onselectend' }) onSelectEnd: EventEmitter

  @Listen('scroll')
  onScroll(_event: UIEvent) {
    if (!this.isMove) {
      this.isMove = true
      this.onSelectStart.emit()
    }
    this.handleSelected()
  }

  @Listen('mouseup')
  @Listen('mouseout')
  @Listen('mouseleave')
  onMouseEnd() {
    if (!this.isMove) return
    this.isMove = false
    this.handleSelected()
  }

  @Listen('touchend')
  onTouchEnd() {
    this.isMove = false
    this.handleSelected()
  }

  componentDidUpdate() {
    if (!this.isInit) {
      this.isInit = true
      const childList = this.el.childNodes
      let idx = 0
      let sum = 0
      for (const index in childList) {
        const item = childList[index] as HTMLElement
        if (this.initialPosition === index || !item || typeof item.offsetHeight !== 'number') {
          break
        }
        sum += item.offsetHeight
        idx++
      }
      this.el.scrollTo({ top: sum })
      if (idx >= childList.length) {
        this.onChange.emit({
          curIndex: this.col,
          selectedIndex: idx - 1
        })
      }
    }
  }

  // 滚动结束自动回到合适的位置
  handleSelected = debounce(() => {
    const childList = this.el.childNodes
    let sum = 0
    let selectedIndex: string = '0'
    for (const index in childList) {
      const item = childList[index] as HTMLElement
      const itemHeight = item.offsetHeight
      if (sum + itemHeight / 2.0 > this.el.scrollTop) {
        selectedIndex = index
        break
      }
      sum += itemHeight
    }

    this.el.scrollTo({
      top: sum,
      behavior: 'smooth'
    })
    this.onChange.emit({
      curIndex: this.col,
      selectedIndex: selectedIndex
    })
    this.onSelectEnd.emit()
  }, 500)

  render() {
    const { paddingVertical = 0 } = this
    return (
      <Host
        class="taro-picker-view-column-container"
        style={{ 'padding-top': `${paddingVertical}px`, 'padding-bottom': `${paddingVertical}px` }}
      />
    )
  }
}
