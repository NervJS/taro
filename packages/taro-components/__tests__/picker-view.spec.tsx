import { h } from '@stencil/core'
import { newSpecPage, SpecPage } from '@stencil/core/testing'

import { PickerView } from '../src/components/picker-view/picker-view'
import { PickerViewColumn } from '../src/components/picker-view/picker-view-column'

describe('PickerView', () => {
  let page: SpecPage

  it('props valid', async () => {
    const date = new Date()
    const years: number[] = []
    for (let i = 1990; i <= date.getFullYear(); i++) {
      years.push(i)
    }
    page = await newSpecPage({
      components: [PickerView, PickerViewColumn],
      template: () => (<taro-view-core>
        <taro-picker-view-core
          indicatorStyle='height: 60px;'
          indicatorClass='test_indicatorClass'
          style='width: 100%; height: 300px; position: absolute; bottom: 0px;'
          maskClass='test_maskClass'
          maskStyle="background-color: rgba(33, 33, 33, 0.5);"
        >
          <taro-picker-view-column-core>
            {years.map(item => (
              <taro-view-core key={item} style='height: 60px; line-height: 60px; text-align: center;'>
                {item}年
              </taro-view-core>
            ))}
          </taro-picker-view-column-core>
        </taro-picker-view-core>
      </taro-view-core>),
    })

    const indicatorEle = page.root?.querySelector('.taro-picker-view-mask-indicator')
    expect(indicatorEle?.classList.contains('test_indicatorClass')).toBe(true)

    const maskTopEle = page.root?.querySelector('.taro-picker-view-mask-top')
    expect(maskTopEle?.classList.contains('test_maskClass')).toBe(true)

    const maskBottomEle = page.root?.querySelector('.taro-picker-view-mask-bottom')
    expect(maskBottomEle?.classList.contains('test_maskClass')).toBe(true)

    const pickerViewColumnEle = page.root?.querySelector('.taro-picker-view-column-container')

    // 模拟滑动
    const startY = 0
    const endY = 200

    pickerViewColumnEle?.dispatchEvent(new Event('mousedown', {
      // @ts-ignore
      clientY: startY
    }))

    pickerViewColumnEle?.dispatchEvent(new Event('mousemove', {
      // @ts-ignore
      clientY: endY
    }))

    expect(page.root).toMatchSnapshot()
  })
})
