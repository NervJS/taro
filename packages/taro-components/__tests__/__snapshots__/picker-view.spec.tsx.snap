// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerView props valid 1`] = `
<taro-picker-view-core class="taro-picker-view-container" style="0: w; 1: i; 2: d; 3: t; 4: h; 5: :; 6:  ; 7: 1; 8: 0; 9: 0; 10: %; 11: ;; 12:  ; 13: h; 14: e; 15: i; 16: g; 17: h; 18: t; 19: :; 20:  ; 21: 3; 22: 0; 23: 0; 24: p; 25: x; 26: ;; 27:  ; 28: p; 29: o; 30: s; 31: i; 32: t; 33: i; 34: o; 35: n; 36: :; 37:  ; 38: a; 39: b; 40: s; 41: o; 42: l; 43: u; 44: t; 45: e; 46: ;; 47:  ; 48: b; 49: o; 50: t; 51: t; 52: o; 53: m; 54: :; 55:  ; 56: 0; 57: p; 58: x; 59: ;;">
  <taro-picker-view-column-core class="taro-picker-view-column-container" col="0" initial-position="0" padding-vertical="0" style="padding-top: 0px; padding-bottom: 0px;">
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1990年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1991年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1992年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1993年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1994年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1995年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1996年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1997年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1998年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      1999年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2000年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2001年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2002年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2003年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2004年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2005年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2006年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2007年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2008年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2009年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2010年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2011年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2012年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2013年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2014年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2015年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2016年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2017年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2018年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2019年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2020年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2021年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2022年
    </taro-view-core>
    <taro-view-core style="0: h; 1: e; 2: i; 3: g; 4: h; 5: t; 6: :; 7:  ; 8: 6; 9: 0; 10: p; 11: x; 12: ;; 13:  ; 14: l; 15: i; 16: n; 17: e; 18: -; 19: h; 20: e; 21: i; 22: g; 23: h; 24: t; 25: :; 26:  ; 27: 6; 28: 0; 29: p; 30: x; 31: ;; 32:  ; 33: t; 34: e; 35: x; 36: t; 37: -; 38: a; 39: l; 40: i; 41: g; 42: n; 43: :; 44:  ; 45: c; 46: e; 47: n; 48: t; 49: e; 50: r; 51: ;;">
      2023年
    </taro-view-core>
  </taro-picker-view-column-core>
  <div class="taro-picker-view-mask-container">
    <div class="taro-picker-view-mask-top test_maskClass" style="background-color: rgba(33, 33, 33, 0.5);"></div>
    <div class="taro-picker-view-mask-indicator test_indicatorClass" style="height: 60px;"></div>
    <div class="taro-picker-view-mask-bottom test_maskClass" style="background-color: rgba(33, 33, 33, 0.5);"></div>
  </div>
</taro-picker-view-core>
`;
